src/kern/stand/mempool.c

```diff
@@ -78,8 +78,14 @@ mempool_create(const mempool_ops_t *ops, size_t objsize, unsigned ncache)
 void
 mempool_destroy(mempool_t *mp)
 {
+	mempool_tls_t *tp;
 	if (!mp)
 		return;
+	tp = tls_get(mp->tls_key);
```

Comment:
This is wrong, because it handles only the current thread. As explained in the other PR -- `pthread_key_create()` destructor handles the thread-local memory. The more problematic part here, though, is that each thread ought to call `mempool_cancel()` before `mempool_destroy()` is called.. otherwise we leak the per-thread memory pools. I think we need to modify `tls_create()` to pass a custom destructor (which would be used instead of `free`).